src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -187,6 +188,23 @@ bool hasGlobalAggregationInQuery(const ASTPtr & query, const ASTSelectQuery & se
     GetAggregatesVisitor(data).visit(query);
     return !data.aggregates.empty() || select_query.groupBy() != nullptr;
 }
+
+Names getShuffleByColumns(const ASTSelectQuery & query)
+{
+    Names shuffle_by_columns;
+
+    if (!query.shuffleBy())
+        return shuffle_by_columns;
+
+    shuffle_by_columns.reserve(query.shuffleBy()->children.size());
+    for (const auto & elem : query.shuffleBy()->children)
+    {
+        if (elem->as<ASTIdentifier>())
+            shuffle_by_columns.push_back(elem->getColumnName());
```

Comment:
we will still do the work i think and it shall be fine just like regular columns.

Reply:
Meaning if end user do something like aliasing a constant or an expression / function as an identifier, we will do the shuffle work as expected.

When the logic reaches here, the ast of the shuffle by can be asserted to ASTIdentifier since it is already validated in the AST parser.